### PR TITLE
[GitHub Actions] - Run RC Validations Workflow 

### PR DESCRIPTION
**Pull request by dannymartinm**
As part of the [Beam Release Process](https://beam.apache.org/contribute/release-guide/), the following PR includes the workflow to run the [run_rc_validation.sh](https://github.com/apache/beam/blob/master/release/src/main/scripts/run_rc_validation.sh) script described in [step 10. Run validations using run_rc_validation.sh](https://beam.apache.org/contribute/release-guide/#run-validations-using-run_rc_validationsh)

Files: 
* Composite action to install common python tools `.github/actions/common-rc-validation/action.yaml`
* Run RC Validations Workflow: `.github/workflows/run_rc_validation.yml`
* Updated `CI.md` file

`* Note`: This PR needs to be merged after [PR#23390](https://github.com/apache/beam/pull/23390) as we are using the `github/actions/setup-gh-cli-linux/action.yml` action to setup GitHub CLI. 

**Work done along with:**  @elink21 

Thank you!

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by dannymartinm** [comment]
R: @damccorm

File: .github/workflows/run_rc_validation.yml [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+          ref: ${{ env.RC_TAG }}
+      - name: Setup GitHub CLI
+        uses: ./.github/actions/setup-gh-cli-linux
+      - name: Set git config
+        run: |
+          git config user.name $GITHUB_ACTOR
+          git config user.email actions@"$RUNNER_NAME".local
+      - name: Create Pull Request
+        run: |
+          git checkout -b ${{env.WORKING_BRANCH}} ${{ env.RC_TAG }} --quiet
```
Comment (damccorm) [inline .github/workflows/run_rc_validation.yml:92]: Can we do some validation to ensure this branch doesn't already exist (theoretically it shouldn't, but that could break the validation)

**Review by damccorm — COMMENTED** [review]
This mostly lgtm, just had a couple comments

**Review by damccorm — APPROVED** [review]
LGTM, thanks!

**Pull request merged**